presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestJsonDateTime.java

```diff
@@ -0,0 +1,195 @@
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder.json;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.prestosql.plugin.kafka.KafkaColumnHandle;
+import io.prestosql.plugin.kafka.encoder.EncoderColumnHandle;
+import io.prestosql.plugin.kafka.encoder.RowEncoder;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.type.DateType;
+import io.prestosql.spi.type.TimeType;
+import io.prestosql.spi.type.TimeWithTimeZoneType;
+import io.prestosql.spi.type.TimestampType;
+import io.prestosql.spi.type.TimestampWithTimeZoneType;
+import io.prestosql.spi.type.Type;
+import io.prestosql.testing.TestingConnectorSession;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+import static io.prestosql.spi.type.DateTimeEncoding.packDateTimeWithZone;
+import static io.prestosql.testing.assertions.Assert.assertEquals;
+
+public class TestJsonDateTime
+{
+    private static final JsonRowEncoderFactory ENCODER_FACTORY = new JsonRowEncoderFactory(new ObjectMapper());
+
+    private static EncoderColumnHandle createColumnHandle(String name, Type type, String dataFormat)
+    {
+        return createColumnHandle(name, type, dataFormat, null);
+    }
+
+    private static EncoderColumnHandle createColumnHandle(String name, Type type, String dataFormat, String formatHint)
+    {
+        return new KafkaColumnHandle(name, type, name, dataFormat, formatHint, false, false, false);
+    }
+
+    private static RowEncoder createRowEncoder(EncoderColumnHandle... columnHandles)
+    {
+        return ENCODER_FACTORY.create(TestingConnectorSession.SESSION, Optional.empty(), ImmutableList.copyOf(columnHandles));
+    }
+
+    @Test
+    public void testCustomFormat()
+    {
+        testDateHelper(18458, "2020-15-07", "custom-date-time", Optional.of("yyyy-dd-MM"));
+        testTimeHelper(1594846503123L, "20:55:03.123", "custom-date-time", Optional.of("kk:mm:ss.SSS"));
+        testTimeWithTZHelper(packDateTimeWithZone(1594846503123L, "UTC"), "20:55:03.123 +0000", "custom-date-time", Optional.of("kk:mm:ss.SSS Z"));
+        testTimestampHelper(1594846503123L, "2020-15-07 20:55:03.123", "custom-date-time", Optional.of("yyyy-dd-MM kk:mm:ss.SSS"));
+        testTimestampWithTZHelper(packDateTimeWithZone(1594846503123L, "UTC"), "2020-15-07 20:55:03.123 +0000", "custom-date-time", Optional.of("yyyy-dd-MM kk:mm:ss.SSS Z"));
+    }
+
+    @Test
+    public void testIso8601Format()
+    {
+        testDateHelper(18458, "2020-07-15", "iso8601", Optional.empty());
+        testTimeHelper(1594846503123L, "20:55:03.123", "iso8601", Optional.empty());
+        testTimeWithTZHelper(packDateTimeWithZone(1594846503123L, "UTC"), "20:55:03.123Z", "iso8601", Optional.empty());
+        testTimestampHelper(1594846503123L, "2020-07-15T20:55:03.123", "iso8601", Optional.empty());
```

Comment:
Use of datatype internal representation here make the test look cryptic. Compute `1594846503123L` using `java.time.*` instead hardcoding it in tests?